Source/DafnyCore/ProofDependencyWarnings.cs

```diff
@@ -81,8 +86,13 @@ public class ProofDependencyWarnings {
         return false;
       }
 
-      // Similarly here
-      if (poDep.ProofObligation is MatchIsComplete or AlternativeIsComplete) {
+      // Some proof obligations occur in a context that the Dafny programmer
```

Comment:
What do you mean by doesn't have control of? How would a `MatchIsComplete` obligation be vacuous? Does that only occur if false is proven before the match is reached?

Reply:
Sometimes Dafny encodes the proof obligations for these things in ways that include assumptions that don't show up in the original Dafny source, including sometimes something as explicit as `assume false`. These lead to warnings about vacuous proofs that the user can't do anything about. The proofs are still sound -- I think the encoding works that way just to make it simpler and more consistent -- but they're structured in such a way that they can involve vacuously-proved assertions.